activemodel/lib/active_model/type/decimal_without_scale.rb

```diff
@@ -6,6 +6,10 @@ class DecimalWithoutScale < BigInteger # :nodoc:
       def type
         :decimal
       end
+
+      def type_cast_for_schema(value)
+        super(value.to_s)
+      end
```

Comment:
This class shouldn't need to implement this method, as it does not return `BigDecimal` objects.